compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -472,7 +472,18 @@ pub(crate) fn check_item_type(tcx: TyCtxt<'_>, def_id: LocalDefId) {
         DefKind::Enum => {
             check_enum(tcx, def_id);
         }
-        DefKind::Fn => {} // entirely within check_item_body
+        DefKind::Fn => {
+            if let Some(name) = tcx.intrinsic(def_id) {
+                intrinsic::check_intrinsic_type(
```

Comment:
out of curiosity, do you know if we are *testing* the check_intrinsic_type logic anywhere in our unit tests? (E.g. via a `#![no_core]` test that adds its own intrinsics?)

Reply:
okay I see b9d8f00118d29f6dd187104004c44bb4280ed911 so there is at least *some* testing